starboard/nplb/player_get_audio_configuration_test.cc

```diff
@@ -0,0 +1,198 @@
+  }
+
+  // Once at least one audio configurations are returned, the return values
+  // shouldn't change during the life time of |player|.
+  if (!initial_configs.empty()) {
+    ASSERT_EQ(initial_configs, configs_after_presenting);
+  }
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
+
+  std::vector<SbMediaAudioConfiguration> configs_after_end;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &configs_after_end));
+  ASSERT_EQ(configs_after_presenting, configs_after_end);
+}
+
+TEST_P(SbPlayerGetAudioConfigurationTest, NoInput) {
+  SbPlayerTestFixture player_fixture(GetParam());
+  if (HasFatalFailure()) {
+    return;
+  }
+
+  std::vector<SbMediaAudioConfiguration> initial_configs;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &initial_configs));
+
+  GroupedSamples samples;
+  if (player_fixture.HasAudio()) {
+    samples.AddAudioSamplesWithEOS(0, 0);
+  }
+  if (player_fixture.HasVideo()) {
+    samples.AddVideoSamplesWithEOS(0, 0);
+  }
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerPresenting());
+
+  std::vector<SbMediaAudioConfiguration> configs_after_presenting;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &configs_after_presenting));
+
+  if (player_fixture.HasAudio()) {
+    ASSERT_FALSE(configs_after_presenting.empty());
+  }
+
+  if (!initial_configs.empty()) {
+    ASSERT_EQ(initial_configs, configs_after_presenting);
+  }
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
+
+  std::vector<SbMediaAudioConfiguration> configs_after_end;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &configs_after_end));
+  ASSERT_EQ(configs_after_presenting, configs_after_end);
+}
+
+TEST_P(SbPlayerGetAudioConfigurationTest, MultipleSeeks) {
+  const int kSamplesToWrite = 8;
+
+  SbPlayerTestFixture player_fixture(GetParam());
+  if (HasFatalFailure()) {
+    return;
+  }
+
+  std::vector<SbMediaAudioConfiguration> initial_configs;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &initial_configs));
+
+  GroupedSamples samples;
+  if (player_fixture.HasAudio()) {
+    samples.AddAudioSamplesWithEOS(0, kSamplesToWrite);
+  }
+  if (player_fixture.HasVideo()) {
+    samples.AddVideoSamplesWithEOS(0, kSamplesToWrite);
+  }
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerPresenting());
+
+  std::vector<SbMediaAudioConfiguration> configs_after_presenting;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &configs_after_presenting));
+
+  if (player_fixture.HasAudio()) {
+    ASSERT_FALSE(configs_after_presenting.empty());
+  }
+
+  if (!initial_configs.empty()) {
+    ASSERT_EQ(initial_configs, configs_after_presenting);
+  }
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Seek(kSbTimeSecond));
```

Comment:
It might be worth checking the value after we reach presenting again (after seek).